### ISSUE TITLE
Custom file associations with globs in the settings "files.associations"

### ISSUE THREAD

**Issue by jmdsg** [vscode-icons/vscode-icons#1727]
- [x] I'm sure this issue is not a _duplicate_.
- [ ] I'm submitting an icon request:

I am having issues when I use a glob in the `"files.associations"` settings of vscode.

I couldn't find anything about it. The only thing I found similar, is this comment mabe by @JimiC in https://github.com/vscode-icons/vscode-icons/issues/1465#issuecomment-366215706 which says vscode does not support glob for icons, however it does support glob for file associations. So i'm not sure that is even related to this.

My workspace configuration looks like this:

```json
{
    "files.associations": {
        "**/imports/ui/*.js": "javascriptreact",
        "**/imports/ui/components/**/*.js": "javascriptreact",
        "**/imports/ui/layouts/**/*.js": "javascriptreact",
        "**/imports/ui/views/**/*.js": "javascriptreact"
    }
}
```

Obviously the configuration works in vscode, i.e. the language is recognized by vscode in those particular files. But the icons are not changed 'dinamically'. When i open a file, the tab shows the icon from javacript:

![image](https://user-images.githubusercontent.com/11861795/45959655-69042f80-bfe8-11e8-98e3-b691c81139f9.png)

But when I select `"Keep Open [Ctrl + k] Enter"` the icon changes:

![image](https://user-images.githubusercontent.com/11861795/45959577-3a865480-bfe8-11e8-8d0c-070040309114.png)

Also, in the explorer the icon changes sometimes. I think this is an issue with the extension, but i've never check the code, so i'm not really sure if it is indeed related to the extension or with vscode itself.

**Comment by JimiC** [comment]
@jmdsg I wonder why aren't you using the `jsx` file extension for those files? This way `vscode` will recognize them without any additional settings.

**Comment by JimiC** [comment]
You don't need any 3rd party extension for `jsx`. It's supported from `vscode` natively. `file.associations` has nothing to do with our extension and we don't control `vscode` behavior on language recognition.

**Comment by jmdsg** [comment]
Who said anything about 3rd party extensions. 

And as far as i see it; its should be either supported completely or not supported at all. And right now its just half way, so it's at least a bug of `vscode-icons`, i.e. this extension should always change the icon when there are custom `file.associations` or never, not sometimes...

**Comment by JimiC** [comment]
Well, the word `extension` can be confusing, whether you refereing to `file extension` or to `vscode` extension. Anyways, there is no bug here. As I explained, we don't control how `vscode` assigns icons when they are referred by their content. In case you don't know it, `vscode` uses a `firstline` regex to determine what's the actual language used in a file despite its file extension. And that's the reason why you see the `js` ison at first and then the `jsx` one.

**Comment by jmdsg** [comment]
Yes, you're right, i just tried seti icons and it behaves the same. The issue is from `vscode`, maybe i add the issue there (because it's annoying). 

Thanks, you can close this issue.
